tests/functional/a/attribute_defined_outside_init.py

```diff
@@ -82,3 +86,10 @@ def prop(self, value):
 class DataClass:
     def __post_init__(self):
         self.a = 42
+
+
+if sys.version_info >= (3, 8):
+    class G(unittest.IsolatedAsyncioTestCase):
```

Comment:
Hey @Pierre-Sassoulas !
Any idea about how we can make this line _pass_ Pylint against 3.7 **and** 3.8+ (3.9 currently in CI) ? 😇 
(It raises `no-member` against 3.7 and when we suppress this warning it then raises `useless-suppression` against 3.9 ☹️)

Reply:
You should probably set the min and max version of the functional test using a pylintrc and multiple files, here's an example with py-version too : https://github.com/PyCQA/pylint/blob/main/tests/functional/ext/typing/redundant_typehint_argument_py310.rc

Reply:
min_pyver=3.10 means the test won't be executed at all under python 3.10, the py-version means that when the interpreter is python 3.11 pylint will still raise python 3.10 related messages.

Reply:
Thanks for the tip ! PR has been updated (CI seems to pass and a changelog fragment has been added).

:wave: